lean/deps/x86_semantics/src/x86_semantics/common.lean

```diff
@@ -1,26 +1,112 @@
 import galois.category.coe1
-
-def unlines : list string → string := list.foldr (λx r, x ++ "\n" ++ r) ""
+import galois.sexpr
 
 -- Library for buidling s expressions
 namespace sexp
 
-def app (s:string) (l:list string) := "(" ++ s ++ list.foldr (λx r, " " ++ x ++ r) ")" l
+------------------------------------------------------------------------
+-- atom
 
--- Pretty print
-def from_list : list string → string
-| [] := "()"
-| (s::l) := app s l
+def symbol := string
 
-def indent : string → string
-  | s := "  " ++ s
+instance : has_append symbol := ⟨string.append⟩
 
-end sexp
+/-- Symbol characters allowed in simple symbols -/
+protected
+def char_symbols : list char :=
+ ['~', '!', '@', '$', '%', '^', '&', '*', '_', '-', '+', '=', '<', '>', '.', '?', '/' ]
+
+/-- Predicate that checks if character is allowed in a simple symbol. -/
+inductive is_symbol_char (c:char) : Prop
+| is_alpha : char.is_alpha c → is_symbol_char
+| is_digit : char.is_digit c → is_symbol_char
+| is_other : c ∈ sexp.char_symbols → is_symbol_char
```

Comment:
This looks copy pasted from SMTLIB.  Do we need all these special symbols in our x86 semantics?

Reply:
It was and I don't think so. Currently the only non-alphanumeric that we use is `$`.

How do you want me to proceed?

Reply:
Maybe just change the list to include '$'.

Reply:
You may want underscore as well.

Reply:
Done.